patches/minecraft/net/minecraft/client/gui/components/toasts/ToastComponent.java.patch

```diff
@@ -0,0 +1,13 @@
+--- a/net/minecraft/client/gui/components/toasts/ToastComponent.java
++++ b/net/minecraft/client/gui/components/toasts/ToastComponent.java
+@@ -103,7 +_,9 @@
+    }
+ 
+    public void m_94922_(Toast p_94923_) {
+-      this.f_94916_.add(p_94923_);
++      var event = net.minecraftforge.client.ForgeHooksClient.onToastAdd(p_94923_);
++      if (event.isCanceled()) return;
++      this.f_94916_.add(event.getToast());
```

Comment:
You are understanding me wrong.
The fired event can not change the toast you passed in.
So it suffices to return a boolean from the hook method, and turn this 4 line patch into a single line patch above the add method.